passes/techmap/extract_reduce.cc

```diff
@@ -50,6 +50,35 @@ struct ExtractReducePass : public Pass
 		log("\n");
 	}
 
+	bool may_reduce(Cell *cell, GateType gt)
+	{
+		if (cell == NULL)
+			return false;
+		bool res = false;
+		if (((cell->type == "$_AND_" && gt == GateType::And) ||
+			 (cell->type == "$_OR_" && gt == GateType::Or) ||
+			 (cell->type == "$_XOR_" && gt == GateType::Xor) ||
+			 (cell->type == "$and" && gt == GateType::And) ||
+			 (cell->type == "$or" && gt == GateType::Or) ||
+			 (cell->type == "$xor" && gt == GateType::Xor)
+			 ))
+		{
+			res = true;
+		}
+		return res;
+	}
+
+	bool is_single_bit(Cell *cell, SigMap sigmap)
+	{
+		bool res = true;
+		auto a = sigmap(cell->getPort("\\A"));
+		auto b = sigmap(cell->getPort("\\B"));
+		auto y = sigmap(cell->getPort("\\Y"));
+		if ((a.size() != 1) || (b.size() != 1) ||(y.size() != 1))
```

Comment:
`((GetSize(a) != 1) || (GetSize(b) != 1) || (GetSize(y) != 1))` would be idiomatic for Yosys code.